messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -388,9 +388,9 @@ private <Q, R> ResultMessage<Publisher<QueryResponseMessage<R>>> interceptAndInv
         DefaultUnitOfWork<StreamingQueryMessage<Q, R>> uow = DefaultUnitOfWork.startAndGet(query);
         return uow.executeWithResult(() -> {
             Object queryResponse = new DefaultInterceptorChain<>(uow, handlerInterceptors, handler).proceed();
-            return query.getResponseType()
-                        .convert(queryResponse)
-                        .map(GenericQueryResponseMessage::asResponseMessage);
+            return Flux.from(query.getResponseType()
```

Comment:
Curious: I'm not entirely sure why this is required. Is this because the `Publisher` returned by the `convert` method doesn't allow a `map` operation?

Reply:
Exactly!

Reply:
Cool, good to know!